public/components/trace_analytics/home.tsx

```diff
@@ -285,7 +290,15 @@ export const Home = (props: HomeProps) => {
   ];
 
   const traceColumnAction = () => {
-    location.assign('#/traces');
+    const tracesPath = '#/traces';
+    if (newNavigation) {
+      coreRefs.application?.navigateToApp(observabilityTracesNewNavID, {
+        path: tracesPath + '?datasourceId=' + dataSourceMDSId[0].id,
+      });
+    } else {
+      location.assign(tracesPath);
```

Comment:
Should `'?datasourceId=' + dataSourceMDSId[0].id` be added to else condition here?

Reply:
Added, it was working because the data source is maintained as the page didn't switch, but having it will be better if the link is shared ect... 
Thanks.